doc/lua-filters.md

```diff
@@ -62,21 +47,36 @@ replace it with a SmallCaps element with the same content.
 To run it, save it in a file, say `smallcaps.lua`, and invoke
 pandoc with `--lua-filter=smallcaps.lua`.
 
+## Why Lua filters over JSON?
+
+[JSONfilters](https://pandoc.org/filters.html) accept a JSON
+representation of the pandoc AST and produce an altered JSON
+representation of the AST. They may be written in any programming
+language, and invoked from pandoc using the `--filter` option.
+
+ However, JSON filters have limitations:
+ 
+ - Writing JSON to stdout and reading it from stdin (twice, once
+on each side of the filter) is inefficient.
+- External dependencies vary between users, and universal JSON
+  filters are not possible.
```

Comment:
I don't think it will be clear to readers what is meant by "universal JSON filters" or why dependency variation is important. I think the original text on this was clearer.